Added Donksoft Toy Vendor (Machine Board) to the Circuit Imprinter [Fixed PR]

Kerbin-Fiber opened the pull request:
:cl: Kerbin_Fiber
add: Donksoft Toy Vendor Boards to the Circuit Imprinter when Illegal tech is high enough.
add: Donksoft Toy Vendor Restocking Units come in Toy Crates from Cargo!
/:cl:
![donksoftrestockingunit](https://user-images.githubusercontent.com/31713423/30152348-6fb2561a-9378-11e7-9923-dec4ce936b17.PNG)

Why? Because it's no fair for the Syndicates to be able to have ~~fun~~ toys and we don't! Once the research from illegal items that were ~~stolen~~ obtained have been deconstructed, the Circuit Imprinter will be able to print out Donksoft Toy Vendor (Machine Board)s!

tgstation-server committed:
Automatic changelog generation for PR #30416 [ci skip]

tgstation-server committed:
Automatic changelog generation for PR #30416 [ci skip]

WJohn commented:
absolute madman 🥇

FantasticFwoosh reviewed (commented):
Ok the donksoft changes are apparent, but please re-fork and try again because opening statement and detail of what you intended to do is 👌  but the commit details are still borked  :dog:

- A easy way to clean a remote is to delete your repo from web-side git via settings, scrolling to the bottom and deleting, then running all they way back to TG github and hitting that fork button, there are fancier high tech ways of doing it but that's a bog standard way of doing it.

So your version of TG station is up to date BEFORE commiting your changes, as you can see there's a lot of extra stuff not relevant to the PR pulled along because your repository stored in your profile is older than TG's current state.

FantasticFwoosh commented:
Can you not remove that changelog to make a cleaner PR?

GunHog commented:
I never looked into how Cargo randomized crates work. I need three stocking units to build the Donksoft vendor, yes?

I glanced at the code, and I understand it to be that it picks five different items from the list to fill the crate. So, I basically have a one-in-three (5 items from a list of 15) chance of getting a donksoft stocking unit. I need a minimum of three crates to get a restocking unit. Given these odds, I imagine that if I actually want one of these things, I am probably going to end up filling the cargo bay with toy crates.

Still, if someone manages to build one or two, I would love to read the story about it.

Armhulen commented:
The donk soft vendor requires a syndicate id, you would have to remove that.

Cyberboss commented on html/changelogs/AutoChangeLog-pr-30416.yml:
```diff
@@ -0,0 +1,4 @@
+author: "Xhuis"
```
Remove this file from your PR

Kerbin-Fiber replied:
It'd help if you told me how to, since I am new to using this Git Tortoise thing.

Kerbin-Fiber committed:
Update AutoChangeLog-pr-30416.yml

ma44 commented:
## As the original creator of this, please for the love of god don't add this in unless you want to drastically buff science and cargo while also making antags and station crew to engage in a powergaming war of who gets this stuff first

'But ma44 don't you like fun stuff?'
Yes fun stuff like this is great, but it shouldn't be **THIS** easily gotten, let's look at the things wrong with this!

- Riot darts, while having a robust supply of **400** reusable riot darts (that can also be used for metal in a autolathe and recycled as well) as well as all of the guns that have riot darts preloaded. This was meant to make the syndicate base never run out of riot darts for whatever reason, and that they also don't have a autolathe

- Alright let's look at the guns because they are obviously the main reason on why this thing is bad for the station

1. The pistol isn't that horribly bad, pretty sure you can get it from somewhere
2. Shotgun is also not that great, only carries 4 shots and requires a pump per shot, usually not enough to stun a person
3. Crossbow, same as shotgun but holds 5 darts
4. The C20r, holds 20 darts, easy to use magazines, fires two darts at a time and fits in a backpack. This is well literally a burst disabler with nearly **infinite** ammo. Remember to duel wield it so you can fire four darts in a goddamm volley. 
5. The l6 saw, 50 rounds and a triple burst. While this is painfully obvious to anyone that have this out, it still has a crapton of suppression it can output. 
6. The SMG, meant to represent the old SABR SMG, comes with **42** rounds and a goddamm **triple burst**. Literally duel wielding this can make you stamina stun someone in a single goddamm volley and suppresses harder than a goddamm l6 saw. As a bonus this can fit in backpacks.

- The origin tech of the circuit board can make RnD unlock large energy crossbows as well as suppressors while only needing to raise the syndicate tech to 2 (usually achievable with low cost TC items, RnD magicry or a TYRANT AI module circuit) Man, so many stuns!

- Op without adrenalins are going to die so hard, even if they do they can be kept down with all of the ammo powergamers can carry. 

# TLDR merge this if you want burst disablers from gun crafting back, but now the entire station can be equipped with this

GunHog commented:
Geez, you could at least let it ruin one round before you drop an accurate and convincing, if somewhat buzzwordy, argument against it. I would have liked to participate in a station-wide deathmatch. At least one... :sad:

ma44 commented:
1. apply for admin
2. spawn this in
3. ?
4. Enjoy deathmatch

The-clown-known-as-Pepe commented:
>this cancer almost every round

no

Kerbin-Fiber committed:
Merge branch 'master' into DonksoftToyVendors

Kerbin-Fiber committed:
Update AutoChangeLog-pr-30416.yml

Kerbin-Fiber committed:
Update AutoChangeLog-pr-30416.yml

Kerbin-Fiber committed:
Update AutoChangeLog-pr-30416.yml

Kerbin-Fiber commented:
Many people don't want this to happen and others do, I suggest we actually try it out before we throw it down the drain, yes it sounds very biased but still we can at least try it. I've taken into account of how many times I've seen Science have Illegal Tech over 1 and it's very rare for someone to actually do it from what I know of. Gunhog did see that you also have a low chance of getting the restocking units from cargo, it kinda cancels out the point of "Taking down operartives with riot foam darts" when you need to be lucky enough to even get the resupply units before the Ops arrive. With the chances so low to even make it (And needing it to be hacked so everyone can use it (Purple light > Yellow)), it seems like it would only be used on extended rounds. Rushing to get the tech for the board is eh but getting the crates from cargo and wasting lot of credits is another factor to take in. Please at least try this before thrown down the drain.

Kerbin-Fiber committed:
Delete AutoChangeLog-pr-30416.yml

Kerbin-Fiber committed:
Merge branch 'DonksoftToyVendors' of https://github.com/Kerbin-Fiber/tgstation into DonksoftToyVendors

Kerbin-Fiber committed:
Comma Problem

One pesky little comma wasn't placed, making errors, put it back in it's place.

ma44 commented:
Any cargonia that tries hard enough can easily get hundreds of thousands of credits just in 20-30 minutes (or maybe just below 100k, as that should be enough for your cool little vendor) 

This would also encourage scientists to go git that illegal tech up so they can enjoy crossbows or some cool pseudo burst disablers.

BlueNothing commented:
Would it be good to encourage scientists to do that?

Kerbin-Fiber committed:
Upped the Illegal Tech requirements

Makes Illegal Tech requirement to get the Donksoft Toy Vendor (Machine board) has been upped from 2 to 4.

Kerbin-Fiber commented:
I upped the tech level for it so it should encourage them to get even more illegal tech if they want it that bad.

PKPenguin321 commented:
I would put the board on a space ruin instead of in R&D, would practically guarantee it couldnt be powergamed while still potentially letting it happen now and then

BlueNothing commented:
Between the high tech level requirements and the non-trivial cost to
cargo... it might or might not. RnD or Mining would definitely have to kick
in hard, and this would almost certainly be instead of null crates to some
significant extent... but how often do people ruin-dive?

On Sep 8, 2017 1:53 AM, "PKPenguin321" <notifications@github.com> wrote:

> I would put the board on a space ruin instead of in R&D, would practically
> guarantee it couldnt be powergamed while still potentially letting it
> happen now and then
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/30485#issuecomment-328006013>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATzMBG5LwXBzTPlh_wRgl24302wmRSixks5sgNZ0gaJpZM4PPb8A>
> .
>

Kerbin-Fiber commented:
so here are the reasons why this is a viable idea:
* You have to get co-operation of TWO departments (Not any greyshitter can get that to happen)
* Science actually has to work to get the tech high enough
* Even if cargo complies, there will be some problems getting all of the restocking units
* When complete, people can make rooms into Foam Force arenas, ~~**Enables FUN**~~

BlueNothing commented:
Re 1: If science is half-decent, cargo is sane, mining comes in with trace
amounts of everything or more, RnD is as good as maxed and cargo can be
assumed to have an arbitrary number of points. To wit, who said greys are
going to be the ones trying to get it? I'd fully expect cargo or sci to be
the ones to try it.

Re 2: There's a good chance that RnD is being done by someone competent,
probably 50/50 odds or better, meaning that unless mining is dead or
decided to screw off and hunt megafauna from the word go, RnD will max out
within 30 minutes at least half the time. From there, cargo has 1-2 hundred
thousand cargo points to play with. Sec loves disablers, cargo would likely
use them in place of shotguns unless megafauna, etc, are expected. So
Cargo's likely rotation: Insulated gloves, toolbelt crate, maybe get some
medical or pin crates, then wait for a big tech disk. From there... expect
either loads of nullcrates or loads of toy crates. That said... the
increased Illegal tech requirement means these will be made only quite
rarely, because if you can *get* them, you can very probably get nullcrates.

On Sep 8, 2017 8:45 AM, "Kerbin-Fiber" <notifications@github.com> wrote:

> so here are the reasons why this is a viable idea:
>
>    - You have to get co-operation of TWO departments (Not any greyshitter
>    can get that to happen)
>    - Science actually has to work to get the tech high enough
>    - Even if cargo complies, there will be some problems getting all of
>    the restocking units
>    - When complete, people can make rooms into Foam Force arenas, *Enables
>    FUN*
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/30485#issuecomment-328093467>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATzMBK2gwAaMooMN8ZJOHPBpnG4Rj67Xks5sgTbwgaJpZM4PPb8A>
> .
>

BlueNothing commented:
That's why I'd suggest Illegal 2. Then again, there's a significant chance
that it'll either pretty much always happen (tech is low, can be achieved
by a sleepwalking RnD specialist), or never happen (if you can get it, you
won't, because cargo is probably emagged if you have Illegal 4 anyway,
meaning that the shuttle will be full of straight nullcrates, since any
traitor with half a brain will get an emag if they can afford it unless
they're going full murderbone).

On Sep 8, 2017 9:13 AM, "Wyatt walavigne@earthlink.net" <wylthhalm@gmail.com>
wrote:

> Re 1: If science is half-decent, cargo is sane, mining comes in with trace
> amounts of everything or more, RnD is as good as maxed and cargo can be
> assumed to have an arbitrary number of points. To wit, who said greys are
> going to be the ones trying to get it? I'd fully expect cargo or sci to be
> the ones to try it.
>
> Re 2: There's a good chance that RnD is being done by someone competent,
> probably 50/50 odds or better, meaning that unless mining is dead or
> decided to screw off and hunt megafauna from the word go, RnD will max out
> within 30 minutes at least half the time. From there, cargo has 1-2 hundred
> thousand cargo points to play with. Sec loves disablers, cargo would likely
> use them in place of shotguns unless megafauna, etc, are expected. So
> Cargo's likely rotation: Insulated gloves, toolbelt crate, maybe get some
> medical or pin crates, then wait for a big tech disk. From there... expect
> either loads of nullcrates or loads of toy crates. That said... the
> increased Illegal tech requirement means these will be made only quite
> rarely, because if you can *get* them, you can very probably get nullcrates.
>
> On Sep 8, 2017 8:45 AM, "Kerbin-Fiber" <notifications@github.com> wrote:
>
>> so here are the reasons why this is a viable idea:
>>
>>    - You have to get co-operation of TWO departments (Not any
>>    greyshitter can get that to happen)
>>    - Science actually has to work to get the tech high enough
>>    - Even if cargo complies, there will be some problems getting all of
>>    the restocking units
>>    - When complete, people can make rooms into Foam Force arenas, *Enables
>>    FUN*
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tgstation/tgstation/pull/30485#issuecomment-328093467>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ATzMBK2gwAaMooMN8ZJOHPBpnG4Rj67Xks5sgTbwgaJpZM4PPb8A>
>> .
>>
>

ma44 commented:
This is going to be one of those features that gets a revert pr in a month because some powergamer decides to build this nearly every round for cool chaos literally everywhere

GunHog commented:
> This is going to be one of those features that gets a revert pr in a month because some powergamer decides to build this nearly every round for cool chaos literally everywhere

Which of course would be you, because powergamers sometimes abuse things as a statement to get them removed!

BlueNothing commented:
Case in point: Vcordie and Telesci.

On Sep 8, 2017 10:13 AM, "GunHog" <notifications@github.com> wrote:

> This is going to be one of those features that gets a revert pr in a month
> because some powergamer decides to build this nearly every round for cool
> chaos literally everywhere
>
> Which of course would be you, because powergamers sometimes abuse things
> as a statement to get them removed!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/30485#issuecomment-328114196>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATzMBJTVMBnz8D1GxWr0BffD4ITR4TG-ks5sgUtvgaJpZM4PPb8A>
> .
>

GunHog commented:
After being educated by @ma44 about riot darts, why not just remove those from the vendor? 

You could have normal darts only affect people with a laser tag vest (or a new Donk vest / toy armor?) so people with 'burst disablers' can only stun other Donksoft/laser tag players. 

Note: Ma44 did not explain how else people acquire riot darts, so I do not know how viable an idea this is.

BlueNothing commented:
Hacked autolathes also make them, but it could be okay if they're Illegal 2
but don't come with ammo?

On Sep 8, 2017 10:28 AM, "GunHog" <notifications@github.com> wrote:

> After being educated by @ma44 <https://github.com/ma44> about riot darts,
> why not just remove those from the vendor?
>
> You could have normal darts only affect people with a laser tag vest (or a
> new Donk vest / toy armor?) so people with 'burst disablers' can only stun
> other Donksoft/laser tag players.
>
> Note: Ma44 did not explain how else people acquire riot darts, so I do not
> know how viable an idea this is.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/30485#issuecomment-328118298>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATzMBKDR2J0ASebt4kl1pBYWhlPtKNd6ks5sgU8fgaJpZM4PPb8A>
> .
>

GunHog commented:
Hacked autolathes? You would have to remove that, then.

BlueNothing commented:
If you make it contingent on wearing a vest, mostly nobody's going to wear such a vest.

Kerbin-Fiber commented:
Alright, I'll change it so that it doesn't give out riot foam darts, or at least to the best of my abilities.

Kerbin-Fiber committed:
Reverts Illegal Tech Level

Reverts Illegal Tech Level back down to 2 instead of 4

Reason: Riot foam darts are being removed from the vendor.

ma44 commented:
The fluke awps donksoft guns start off with riot darts. Also note that some of the guns can also prespawn with riot darts.

Ratvar commented:
animated/vendorrisen donksoft should shoot people with riot darts down and glomp up in close range
totally

juanototo2 commented:
Dude the Syndicate lavabase needs riot darts if you can just make it a subtype without the riots instead jeez

Kerbin-Fiber commented:
That's what I'm doing. I'm removing all riot foam darts from the vendor by making it a subtype.

ma44 commented:
May as well rename this pr 'removes riot darts from the autolathe while adding in a cool toy gun vendor that doesn't have riot darts'

Kerbin-Fiber commented:
I said I'm not touching the Autolathe, this PR is for the Vendor only.
